src/components/organisms/NeynarUserDropdown/index.tsx

```diff
@@ -0,0 +1,172 @@
+    style?: React.CSSProperties;
+    placeholder?: string;
+    disabled?: boolean;
+    viewerFid?: number;
+}
+
+async function fetchUsersByQuery({
+  q,
+  viewerFid,
+  client_id
+}: {
+  q: string;
+  viewerFid?: number;
+  client_id: string;
+}): Promise<User[] | null> {
+  try {
+    let requestUrl = `${NEYNAR_API_URL}/v2/farcaster/user/search?q=${q}&limit=5${viewerFid ? `&viewer_fid=${viewerFid}` : ''}&client_id=${client_id}`;
+    const response = await customFetch(requestUrl);
+    const data = await response.json();
+    return data?.result?.users || [];
+  } catch (error) {
+    console.log("Error fetching users by query", error);
+    return null;
+  }
+}
+
+export const NeynarUserDropdown: React.FC<NeynarUserDropdownProps> = ({
+ value: propValue,
+ onChange,
+  style = {},
+  placeholder = 'Enter FIDs or usernames',
+  disabled = false,
+  viewerFid
+}) => {
+    const { client_id } = useNeynarContext();
+    const [internalValue, setInternalValue] = useState(propValue || '');
+    const [currentValue, setCurrentValue] = useState<string>('');
+    const [showDropdown, setShowDropdown] = useState<boolean>(false);
+    const [users, setUsers] = useState<User[]>([]);
+    const inputRef = useRef<HTMLInputElement>(null);
+  
+    useEffect(() => {
+      if (propValue !== undefined) {
+        setInternalValue(propValue);
+      }
+    }, [propValue]);
+
+    useEffect(() => {
+      const values = internalValue.split(',');
+      if (!values[values.length -1]) {
+       setCurrentValue('');
+        return;
+      }
+      setCurrentValue(values[values.length - 1].trim());
+    }, [internalValue]);
+  
+    useEffect(() => {
+        const shouldFetchUsers = currentValue !== '' && !/^\d+$/.test(currentValue);
+        if (shouldFetchUsers) {
+          fetchUsers(currentValue);
+        } else {
+          setShowDropdown(false);
+        }
+      }, [currentValue]);
+    
+    const fetchUsers = async (query: string) => {
+      const fetchedUsers = await fetchUsersByQuery({ q: query, viewerFid, client_id });
+      if (fetchedUsers) {
+        setUsers(fetchedUsers);
+        setShowDropdown(true);
+      }
+    };
+  
+    const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+      const newValue = e.target.value;
+      setInternalValue(newValue);
+      onChange && onChange(newValue);
+    };
+  
+    const handleUserSelect = (user: User) => {
+      let values = internalValue.split(',');
+      values[values.length - 1] = user.fid.toString();
+      const newValue = values.join(',');
+      setInternalValue(newValue);
+      onChange && onChange(newValue);
+      setCurrentValue('');
+      setShowDropdown(false);
+    };
+  
+    const handleInputFocus = () => {
+      if (currentValue && isNaN(Number(currentValue))) {
+        setShowDropdown(true);
+      }
+    };
+  
+    const handleInputBlur = () => {
+      setTimeout(() => setShowDropdown(false), 200);
+    };
+  
+    return (
```

Comment:
could you move the actual dropdown component to the `molecules` folder(eg. `UserDropdown.tsx` or `Dropdown.tsx`) -- usually we've had organisms fetch data and pass them down to molecules to follow the atomic design structure